### PR TITLE
[ProteomicsLFQ] allow non-probabilistic scores for basic inference

### PR DESCRIPTION
**Pull request by jpfeuffer**
Not recommended though. @jonasscheid

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.openms.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

### CODE REVIEW

**Comment by jonasscheid** [comment]
Thanks for the fast PR @jpfeuffer 🙏🏼 ! Would you recommend bayesian inference for peptide quantification (only) or is `aggregated` also sufficient? Or does it not matter if I set proteinFDR = 1 since I'm only interested in peptide quantities

**Comment by jpfeuffer** [comment]
Protein FDR 1 will unfortunately not disable inference. But if you are not interested in protein scores I would suggest "aggregated" because it is much faster.

I can think about allowing no inference but have to check how this propagates through the rest of the program.

File: src/topp/ProteomicsLFQ.cpp [hunk 1]
```diff
@@ -848,8 +848,15 @@ class ProteomicsLFQ :
     }
     catch(Exception::MissingInformation&)
     {
-      OPENMS_LOG_FATAL_ERROR << "ProteomicsLFQ expects a Posterior Error Probability score in all Peptide IDs. ID file: " << id_file_abs_path << endl;
-      return ExitCodes::INCOMPATIBLE_INPUT_DATA;
+      if (getStringOption_("protein_inference") == "bayesian")
+      {
+        OPENMS_LOG_FATAL_ERROR << "ProteomicsLFQ with Bayesian inference expects a Posterior Error Probability score in all Peptide IDs. Affected ID file: " << id_file_abs_path << endl;
```
Comment (poshul) [inline src/topp/ProteomicsLFQ.cpp]:
```suggestion
        OPENMS_LOG_FATAL_ERROR << "ProteomicsLFQ with Bayesian inference expects a Posterior Error Probability score in all Peptide IDs. Effected ID file: " << id_file_abs_path << endl;
```

File: src/topp/ProteomicsLFQ.cpp [hunk 2]
```diff
@@ -848,8 +848,15 @@ class ProteomicsLFQ :
...
+        return ExitCodes::INCOMPATIBLE_INPUT_DATA;
+      } 
+      else
+      {
+        OPENMS_LOG_WARN << "A probabilistic score is recommended for protein inference in ProteomicsLFQ but none was found that applies to all IDs. Affected ID file: " << id_file_abs_path << endl;
```
Comment (poshul) [inline src/topp/ProteomicsLFQ.cpp]:
```suggestion
        OPENMS_LOG_WARN << "A probabilistic score is recommended for protein inference in ProteomicsLFQ but none was found that applies to all IDs. Effected ID file: " << id_file_abs_path << endl;
```

**Review by poshul — CHANGES_REQUESTED** [review]

**Comment by jonasscheid** [comment]
Thanks for looking into it @jpfeuffer 🚀 !

**Comment by timosachsenberg** [comment]
should we only aloow this if --force flag is set?

**Comment by jpfeuffer** [comment]
Sure, fine with me. Not sure if this is even necessary anymore. I think Jonas found a workaround.

**Comment by timosachsenberg** [comment]
ok then let's close and reopen if required

**Pull request closed without merging**